extensions/resume.md

```diff
@@ -0,0 +1,233 @@
+---
+title: IRCv3 `resume` extension
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: Daniel Oakley
+    period: 2017-2018
+    email: daniel@danieloaks.net
+---
+## Introduction
+
+Occasionally, clients disconnect from IRC. What normally happens is that the client connects with a different nickname, joins all their old channels, waits for the old connection to time out (or manually kills it using services), and then changes back to their original nickname.
+
+The `resume` feature vastly simplifies this form of reconnection, for both the client that's reconnecting and clients joined to the same channels. In addition, this feature allows servers to either send missing chat history to the reconnecting client, or make other clients aware of just how much history has been lost.
+
+
+## Architecture
+
+This feature is enabled using the `draft/resume-0.2` capability, introduces the `RESUME` command, and uses the messages described below to convey state about the reconnection process and reconnecting clients.
+
+### Resumption Token
+
+Upon indicating they support session resumption, clients are provided with a resumption token. This token is used to authenticate future `RESUME` attempts.
+
+Servers MUST ensure that this token is long, cryptographically secure, and not able to be easily brute-forced as this token will protect the security of clients' information. Resumption tokens are case-sensitive, and clients MUST treat them as opaque values.
+
+The token MUST NOT contain any spaces and MUST NOT start with a colon character `(:)`, as these would break the protocol framing. There is no sort of encoding or decoding defined for the token.
+
+### Capabilities
+
+The `draft/resume-0.2` capability is advertised by servers that support resuming connections.
+
+When this capability is negotiated, the server sends the client the resumption token, which is used to authenticate future connection resumptions. In addition, servers MAY extend the "ping timeout" length for that client (with the expectation that if the client disconnects, it will try to resume its connection as described here).
+
+Servers MAY ONLY generate and provide a resumption token to a client when the client negotiates the `draft/resume-0.2` capability.
+
+Capability negotiation example:
+
+    C: CAP LS
+    S: CAP * LS :multi-prefix draft/resume-0.2
+    C: CAP REQ :draft/resume-0.2
+    S: CAP * ACK :draft/resume-0.2
+    S: RESUME TOKEN JKbAypzFiovffzuD8VEfcs6bOLrXsSenxsyZNt87Bb0YaQlyOzuQciz6cB2R
+
+### Commands
+
+#### `RESUME` Command
+
+This command MAY ONLY be sent by a client during connection registration, and indicates that the client wishes to resume their session.
+
+    RESUME <oldnick> <token> [timestamp]
```

Comment:
why do we need `<oldnick>`? the server already knows what old nick the token is tied to. seems pointless to make the client keep this extra piece of information and there's scenarios in which the nick of the old connection could have changed while the client was pinging out which would cause the new connection's `RESUME` to fail

Reply:
`<oldnick>` no longer required after some good security discussion back-and-forth. Thanks for the yell!